src/hooks/useLocalStorage/index.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react'
+
+interface useLocalStorageHook {
+  setLocalValue: (value: string) => void
+  storedLocalValue: string
+  removeLocalItem: () => void
+}
+export const useLocalStorage = (
+  key: string,
+  initialValue: string
+): useLocalStorageHook => {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedLocalValue, setStoredLocalValue] = useState(() => {
+    if (typeof window === 'undefined') {
+      return initialValue
+    }
+    try {
+      // Get from local storage by key
+      const item = window.localStorage.getItem(key)
+      // Parse stored json or if none return initialValue
+      return item ? `${item}` : initialValue
+    } catch (error) {
+      // If error also return initialValue
+      return `${initialValue}`
+    }
+  })
+  // Return a wrapped version of useState's setter function that ...
+  // ... persists the new value to localStorage.
+  const setLocalValue = (value: any) => {
+    try {
+      // Allow value to be a function so we have same API as useState
+      const valueToStore =
+        value instanceof Function ? value(storedLocalValue) : value
+      // Save state
+      setStoredLocalValue(valueToStore)
+      // Save to local storage
+      if (typeof window !== 'undefined') {
+        window.localStorage.setItem(key, valueToStore)
+      }
+    } catch (error) {
+      throw error
```

Comment:
what can be done for this error? is it automatically reported in Sentry so that's it?

Reply:
I've downgraded it to a console.warn -- there's nothing to do, correct. Any error setting the localstorage value will only degrade the saving of responses locally, no impact on the session itself so no need to throw an error